temporalio/worker.py

```diff
@@ -0,0 +1,1107 @@
+                    )
+            except Exception as inner_err:
+                temporalio.activity.logger.exception(
+                    f"Exception handling failed, original error: {err}, failure: {inner_err}",
+                    extra={"error": err},
+                )
+                completion.result.Clear()
+                completion.result.failed.failure.message = (
+                    f"Failed building exception result: {inner_err}"
+                )
+
+        # If there was a last heartbeat task, we have to wait on it. We mark the
+        # activity as already done to make sure that the heartbeater doesn't try
+        # to cancel it if it fails.
+        running_activity.done = True
+        if running_activity.last_heartbeat_task:
+            task = running_activity.last_heartbeat_task
+            running_activity.last_heartbeat_task = None
+            try:
+                await task
+            except:
+                # Should never happen because it's trapped in-task
+                temporalio.activity.logger.exception(
+                    "Final heartbeat task didn't trap error"
+                )
+        # Send task completion to core
+        del self._running_activities[task_token]
+        try:
+            await self._bridge_worker.complete_activity_task(completion)
+        except Exception as err:
+            temporalio.activity.logger.exception(
+                f"Failed completing activity task: {err}", extra={"error": err}
+            )
+
+
+class WorkerConfig(typing_extensions.TypedDict):
+    """TypedDict of config originally passed to :py:meth:`Worker`."""
+
+    client: temporalio.client.Client
+    task_queue: str
+    activities: Mapping[str, Callable]
+    activity_executor: Optional[concurrent.futures.Executor]
+    interceptors: Iterable[Interceptor]
+    max_cached_workflows: int
+    max_outstanding_workflow_tasks: int
+    max_outstanding_activities: int
+    max_outstanding_local_activities: int
+    max_concurrent_wft_polls: int
+    nonsticky_to_sticky_poll_ratio: float
+    max_concurrent_at_polls: int
+    no_remote_activities: bool
+    sticky_queue_schedule_to_start_timeout: timedelta
+    max_heartbeat_throttle_interval: timedelta
+    default_heartbeat_throttle_interval: timedelta
+    graceful_shutdown_timeout: timedelta
+    shared_state_manager: Optional[SharedStateManager]
+    type_hint_eval_str: bool
+
+
+@dataclass
+class _ActivityDefinition:
+    name: str
+    fn: Callable[..., Any]
+    arg_types: Optional[List[Type]]
+    ret_type: Optional[Type]
+
+
+@dataclass
+class _RunningActivity:
+    task: Optional[asyncio.Task]
+    cancelled_event: temporalio.activity._CompositeEvent
+    worker_shutdown_event: temporalio.activity._CompositeEvent
+    last_heartbeat_task: Optional[asyncio.Task]
+    sync: bool
+    done: bool = False
+    # Increased each heartbeat, heartbeats not at this seq are ignored
+    current_heartbeat_seq: int = 10
```

Comment:
Why start at 10?
Also please expand the explanation why this is needed here please.

Reply:
No real reason to start at 10 or 0 or 1. I will add comments, but basically like most sequence capturing during sync/async boundaries, it's to prevent out-of-order processing.

Reply:
Changed to 1 and documented its purpose (though it might change)